README.md

```diff
@@ -58,7 +58,7 @@ Here's what you can configure:
 Hangfire.Console provides extension methods on `PerformContext` object, 
 hence you'll need to add it as a job argument. 
 
-**NOTE**: Like `IJobCancellationToken`, `PerformContext` is a special argument type which Hangfire will substitute automatically. You should pass `null` when enqueuing a job.
+**NOTE**: Like `IJobCancellationToken`, `PerformContext` is a special argument type which Hangfire will substitute automatically. You should pass `null` when enqueuing a job.  It is not currently possible to use constructor injection for the PerformContext in ASP.NET Core.
 
```

Comment:
You can actually do that with a custom job activator:
```c#
public class InjectContextJobActivator : JobActivator
{
    private readonly IServiceScopeFactory _scopeFactory;
    
    public InjectContextJobActivator(IServiceScopeFactory scopeFactory)
    {
        _scopeFactory = scopeFactory ?? throw new ArgumentNullException(nameof(scopeFactory));
    }

    public override JobActivatorScope BeginScope(PerformContext context)
    {
        return new Scope(context, _scopeFactory.CreateScope());
    }

    private class Scope : JobActivatorScope, IServiceProvider
    {
        private readonly PerformContext _context;
        private readonly IServiceScope _scope;

        public Scope(PerformContext context, IServiceScope scope)
        {
            _context = context ?? throw new ArgumentNullException(nameof(context));
            _scope = scope ?? throw new ArgumentNullException(nameof(scope));
        }
        
        public override object Resolve(Type type)
        {
            return ActivatorUtilities.GetServiceOrCreateInstance(this, type);
        }

        object IServiceProvider.GetService(Type serviceType)
        {
            if (serviceType == typeof(PerformContext))
                return _context;
            return _scope.ServiceProvider.GetService(serviceType);
        }
    }
}
```

Just register it as a service in `ConfigureServices`:
```c#
services.AddSingleton<JobActivator, InjectContextJobActivator>();
```

Reply:
Just for future readers that have copied this code basically blindly, i just found out that because the serviceScope is not disposed this will cause a memory leak and possible myriad of other issues if your transient/scoped services are not cleaned up properly.

Reply:
I.e. add:

``` c#
/// <inheritdoc />
public override void DisposeScope()
{
    _scope.Dispose();
}
```